pep-9999.rst

```diff
@@ -0,0 +1,506 @@
+represent requirements rather then :pep:`508` strings:
+
+- Easy initial validation via the TOML syntax.
+
+- Easy secondary validation using a schema, for example a `JSON Schema`_.
+
+- Potential for users to guess the keys of given features, rather than
+  memorising a syntax.
+
+- Users of multiple other popular languages may already be familiar with the
+  TOML syntax.
+
+- TOML directly represents the same data structures as in JSON, and therefore a
+  sub-set of Python literals, so users can understand the hierarchy and type of
+  value
+
+.. _JSON Schema: https://json-schema.org/
+
+
+Rationale
+=========
+
+Most of this is taken from discussions in the `PEP 621 dependencies topic`_.
+This has elements from `Pipfile`_, `Poetry`_, `Dart's dependencies`_ and
+`Rust's Cargo`_. A `comparison document`_ shows advantages and disadvantages
+between this format and :pep:`508`-style specifiers.
+
+In the specification of multiple requirements with the same distribution name
+(where environment markers choose the appropriate dependency), the chosen
+solution is similar to `Poetry`_'s, where an array of requirements is allowed.
+
+The direct-reference keys closely align with and utilise pep:`610` and
+:pep:`440` as to reduce differences in the packaging ecosystem and rely on
+previous work in specification.
+
+.. _PEP 621 dependencies topic: https://discuss.python.org/t/pep-621-how-to-specify-dependencies/4599
+.. _Pipfile: https://github.com/pypa/pipfile
+.. _Poetry: https://python-poetry.org/docs/dependency-specification/
+.. _Dart's dependencies: https://dart.dev/tools/pub/dependencies
+.. _Rust's Cargo: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html
+.. _comparison document: https://github.com/uranusjr/packaging-metadata-comparisons/blob/master/topics/dependency-entries.md
+
+
+Specification
+=============
+
+As in :pep:`621`, if metadata is improperly specified then tools MUST raise an
+error. The metadata MUST conform to the `toml`_ specification.
+
+To reduce confusion with this document being a specification for specifying
+dependencies, the word "requirement" is used to mean a :pep:`508` dependency
+specification.
+
+The following tables are added to the added to the ``project`` table specified
+in :pep:`621`.
+
+.. _toml: https://toml.io/
+
+dependencies
+------------
+
+Format: table
+
+The keys inside this table are the names of the required distribution. The
+values can have one of the following types:
+
+- string: the requirement is defined only by a version requirement, with same
+  specification as ``version`` in the requirement table, except allowing the
+  empty string ``""`` to place no restriction on the version.
+
+- table: a requirement table.
+
+- array: an array of requirement tables. It is an error to specify an empty
+  array ``[]`` as a value.
+
+.. _requirement-spec:
+
+Requirement table
+^^^^^^^^^^^^^^^^^
+
+The keys of the requirement table are as follows (all are optional):
```

Comment:
What about keys which are not listed? Are they to be ignored, raise a warning, or raise an error?